src/foam/core/FObject.java

```diff
@@ -296,6 +296,10 @@ default void append(java.lang.StringBuilder sb) {
   }
 
   default FObject copyFrom(FObject obj) {
+    return copyFrom_(obj, obj.getClass());
+  }
+
+  default <T extends FObject> T copyFrom_(FObject obj, Class<T> type) {
```

Comment:
What is the reason for this?

Reply:
It allows overriding copyFrom if the argument object is a type of interest eg. A.copyFrom(B) to behave differently than the default FObject.copyFrom.

Another option is to overload copyFrom in class A to accept B as it argument but it doesn't work when used in CopyFromDAO because it only has FObject to work with https://github.com/kgrgreer/foam3/blob/development/src/foam/dao/CopyFromDAO.js#L62.

Reply:
Not needed, can directly override copyFrom in subclasses.